### PR TITLE
Add documentation to dma_channel_abort regarding errata RP2040-E13

### PR DESCRIPTION
**Pull request by kilograham**
See https://github.com/raspberrypi/pico-feedback/issues/224

Note this is a documentation only fix, as `dma_channel_abort` does not have enough information to efficiently workaround the issue on the callers behalf.

### CODE REVIEW

File: src/rp2_common/hardware_dma/include/hardware/dma.h [hunk 1]
```diff
@@ -473,6 +473,32 @@ static inline void dma_channel_start(uint channel) {
...
+ * transfers will trigger a completion interrupt once they complete.
+ *
+ * The effect of this is that you \em may see a spurious completion interrupt
+ * on the channel as a result of calling this method.
+ *
+ * The calling code should be sure to ignore a completion IRQ as a result of this method. This may
+ * not require any additional work, as aborting a channel which may be about to complete, when you have a completion
+ * IRQ handler registered, is inherently race-prone, and so code is likely needed to disambiguate the two occurrences.
+ *
+ * If that is not the case, but you do have an channel completion IRQ handler registered, you can simply
```
Comment (lurch) [inline src/rp2_common/hardware_dma/include/hardware/dma.h:488]: "an channel completion" -> "a channel completion"

**Comment by Wren6991** [comment]
I fixed the typo pointed out by @lurch. I also made another change, which might be more contentious, so feel free to back out: poll BUSY rather than ABORT to check for ABORT completion, to guarantee the channel is returned to a quiescent state before the function exits. ABORT clears prematurely due to the erratum (in fact this *is* the erratum).

The difference in timing is the time taken for the in-flight transfers to drain from the bus FIFOs, which should be a few cycles.

**Review by Wren6991 — DISMISSED** [review]
Approval for the docs updates, @kilograham please merge if you are happy with the ABORT -> BUSY polling change

**Comment by kilograham** [comment]
Does the ABORT->BUSY change have an effect on what happens if the channel is disabled?

**Comment by Wren6991** [comment]
No. BUSY clears once the in-flight and not-yet-issued transfer counts both reach 0. The not-yet-issued count is cleared by the abort, and the EN bit being cleared does not prevent in-flight transfers from completing.

File: src/rp2_common/hardware_dma/include/hardware/dma.h [hunk 2]
```diff
@@ -473,14 +473,40 @@ static inline void dma_channel_start(uint channel) {
  *
  * Function will only return once the DMA has stopped.
  *
+ * Note that due to to errata RP2040-E13, aborting a channel which has transfers
```
Comment (bruelltuete) [inline src/rp2_common/hardware_dma/include/hardware/dma.h:476]: Another typo: "to" is twice.

**Comment by kilograham** [comment]
Note I force merged this as the only change after approval was a type in a comment

**Pull request merged**